independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/ResteasyReactiveRequestContext.java

```diff
@@ -132,6 +133,9 @@
      */
     private List<UriMatch> matchedURIs;
 
+    private Map<String, String> responseHeaders;
+    private Integer responseStatus;
```

Comment:
I am not to fond of adding new fields here to be honest...

Reply:
Since we only have to change headers and statuses if everything is ok, I don't see a more elegant solution than passing them to the ResteasyReactiveRequestContext, and change them if successful

Reply:
I understand the rationale here, but I don't want to set the precedent of adding new fields for little gain.
Furthermore, by looking at the class, one would not really know when these fields are set, when it's safe to use them and when they actually affect the final result.

Reply:
In theory, I can add annotation checking to AsyncReturnTypeScanner and pass values to the handler, but this will spawn a chain of new fields though the life cycle of annotation will be more clear

Reply:
Given the fact that I completely certain that when I look at this code down the line I will not know when these fields are used, it's going to be a hard no to adding these. We should not have to add custom fields each time we add peripheral functionality.
Please consider alternatives.

Reply:
But again since we want to support all types, not only async that will not work